src/NeinLinq/InjectLambdaMetadata.cs

```diff
@@ -33,8 +34,8 @@ public static InjectLambdaMetadata Create(PropertyInfo property)
         var metadata = InjectLambdaAttribute.Provider(property)
             ?? InjectLambdaAttribute.Provider(property.GetGetMethod(true)!);
 
```

Comment:
We can reduce allocations in moving this right here, I think:

```cs
// property with a setter but no metadata should not be injectable as that's just a normal property
if (property.GetSetMethod(true) is not null && metadata is null)
    return None; // static readonly None = new InjectLambdaMetadata(false, null);
```

Thus, we don't allocate a Lazy object for every property, which has no metadata anyway. And, in using a single static readonly instace "None" we'd even allocate this case only once.

Switching from `Lazy<Func<Expression?, LambdaExpression?>?>` to `Lazy<Func<Expression?, LambdaExpression?>>?` should make this work.